### ISSUE TITLE
Increasing and eventually GPU OOM when running chitchat example script

### ISSUE THREAD

**Issue by gmkim90** [facebookresearch/ParlAI#257]
I am running chitchat example sciprt with following command.

python examples/train_model.py -m seq2seq -t opensubtitles -hs 1024 -nl 2 -dr 0.3 -bs 16 --dict-file 'data/OpenSubtitles/dict'

Used GPU memory starts from 4GB, and keep increasing until comparing prediction & label (See log below), and at some point, it causes GPU Out-of-Memory (in 12GB, Titan X). For me, it seems that GPU memory is cleared (or garbagecollect) at every end of prediction.

Do you recommend something to fix this issue?
FYI, I remove long sentence (more than 30 words) from training data to reduce RNN sequence length.

[LOG]
 time:628s parleys:71 total_exs:1136 ] {'total': 16, 'accuracy': 0, 'f1': 0.03409, 'hits@k': {1: 0, 5: 0, 10: 0, 50: 0, 100: 0}}
[ time:637s parleys:72 total_exs:1152 ] {'total': 16, 'accuracy': 0, 'f1': 0, 'hits@k': {1: 0, 5: 0, 10: 0, 50: 0, 100: 0}}
[ time:646s parleys:73 total_exs:1168 ] {'total': 16, 'accuracy': 0, 'f1': 0, 'hits@k': {1: 0, 5: 0, 10: 0, 50: 0, 100: 0}}
[ time:655s parleys:74 total_exs:1184 ] {'total': 16, 'accuracy': 0, 'f1': 0.005435, 'hits@k': {1: 0, 5: 0, 10: 0, 50: 0, 100: 0}}
[ time:664s parleys:75 total_exs:1200 ] {'total': 16, 'accuracy': 0, 'f1': 0, 'hits@k': {1: 0, 5: 0, 10: 0, 50: 0, 100: 0}}
[ time:673s parleys:76 total_exs:1216 ] {'total': 16, 'accuracy': 0, 'f1': 0, 'hits@k': {1: 0, 5: 0, 10: 0, 50: 0, 100: 0}}
**prediction: How How How away How __NULL__ __NULL__ __NULL__ __NULL__ How __NULL__ __NULL__ __NULL__ __NULL__ __NULL__ __NULL__
label: Right ? __END__ __NULL__ __NULL__ __NULL__ __NULL__ __NULL__ __NULL__ __NULL__ __NULL__ __NULL__ __NULL__ __NULL__ __NULL__ __NULL__**
[ time:682s parleys:77 total_exs:1232 ] {'total': 16, 'accuracy': 0, 'f1': 0, 'hits@k': {1: 0, 5: 0, 10: 0, 50: 0, 100: 0}}
**prediction: Let Right Right . . Right . . __END__ __NULL__ __NULL__ __NULL__ __NULL__ __NULL__ __NULL__ __NULL__ __NULL__ __NULL__
label: - This is not a good situation . __END__ __NULL__ __NULL__ __NULL__ __NULL__ __NULL__ __NULL__ __NULL__ __NULL__ __NULL__**
[ time:692s parleys:78 total_exs:1248 ] {'total': 16, 'accuracy': 0, 'f1': 0, 'hits@k': {1: 0, 5: 0, 10: 0, 50: 0, 100: 0}}
**prediction: He [ He take __END__ Go __NULL__ take __NULL__ ] Crowd __NULL__ __NULL__ __NULL__ __NULL__
label: Is she OK ? __END__ __NULL__ __NULL__ __NULL__ __NULL__ __NULL__ __NULL__ __NULL__ __NULL__ __NULL__ __NULL__**

**Comment by Henry-E** [comment]
yeh I remember when I was training one of the chitchat models I kept having to reduce batch size till the memory errors stopped. Not sure what the actual underlying issue is

**Comment by alexholdenmiller** [comment]
Thanks for flagging! Added a few changes here #269, I think it should prevent it but I'll try your command as well and see if I run out of GPU RAM.

**Comment by alexholdenmiller** [comment]
so far so good on my run, I think this is resolved
